src/field_theory/chevalley_warning.lean

```diff
@@ -88,7 +88,7 @@ end
 
 variables [decidable_eq K] [decidable_eq σ]
 
-/-- The Chevalley–Warning theorem.
+/-- The **Chevalley–Warning theorem**, finitary version.
```

Comment:
Does adding a fintype version (replacing `{x : σ → K // ∀ i ∈ s, eval x (f i) = 0}` with `{x : σ → K // ∀ i, eval x (f i) = 0}`) help the binary version at all?

Reply:
It will shave one lemma off the `simp only ... at key`

Reply:
Actually, it's worse, because the current proof uses the "useless" `i ∈ {tt, ff}` to case on `i`.

Reply:
No it doesn't, `bool.forall_bool` does that!